docs/testing/playwright/01-overview.md

```diff
@@ -0,0 +1,66 @@
+---
+title: Playwright Overview
+sidebar_label: Overview
+---
+
+:::note
+The Stencil Playwright adapter is designed to only work with **version 4.13.0 and higher** of Stencil!
+:::
+
+[Playwright](https://playwright.dev/) is an automated end-to-end testing framework built to run on all modern browser engines and operating systems.
+Playwright leverages the DevTools protocol to provide reliable tests that run in actual browsers.
+
+## Set Up
+
+To add Playwright to an existing Stencil project, leverage the [Stencil Playwright testing adapter](https://www.npmjs.com/package/@stencil/playwright). This
+is a tool built by the Stencil team to help Stencil and Playwright work better together. The best part is you'll write your tests using the same APIs
+as defined and documented by Playwright. So, be sure to [check out their documentation](https://playwright.dev/docs/writing-tests) for help writing your first tests!
+
+To install the Stencil Playwright adapter in an existing Stencil project, follow these steps:
+
+1. Install the necessary dependencies:
+
+   ```bash npm2yarn
+   npm i @stencil/playwright @playwright/test --save-dev
+   ```
+
+1. Install the Playwright browser binaries:
+
+   ```bash
+   npx playwright install
+   ```
+
+1. Create a Playwright config at the root of your Stencil project:
+
+   ```ts title="playwright.config.ts"
+   import { expect } from '@playwright/test';
+   import { matchers, createStencilPlaywrightConfig } from '@stencil/playwright';
+
+   // Add custom Stencil matchers to Playwright assertions
+   expect.extend(matchers);
+
+   export default createStencilPlaywrightConfig({
+     // Overwrite Playwright config options here
+   });
+   ```
+
+   The `createStencilPlaywrightConfig()` is a utility that will create a default Playwright configuration based on your project's Stencil config. Read
+   more about how to use this utility in the [API docs](./03-api.md#createstencilplaywrightconfig-function).
+
+1. Ensure the Stencil project has a [`www` output target](../../output-targets/www.md). Playwright relies on pre-compiled output running in a dev server
+   to run tests against. When using the `createStencilPlaywrightConfig()` helper, a configuration for the dev server will be automatically created based on
+   the Stencil project's `www` output target config and [dev server config](../../config/dev-server.md). If no `www` output target is specified,
+   tests will not be able to run.
```

Comment:
do I need any specific `devServer` config? I have this Stencil config:

```ts
import { Config } from '@stencil/core';

export const config: Config = {
  namespace: 'test-playwright',
  outputTargets: [
    {
      type: 'dist',
      esmLoaderPath: '../loader',
    },
    {
      type: 'dist-custom-elements',
    },
    {
      type: 'docs-readme',
    },
    {
      type: 'www',
      serviceWorker: null, // disable service workers
    },
  ],
  testing: {
    browserHeadless: "new",
  },
  devServer: {
  }
};
```

and when I do `npx playwright` I just see a timeout, like this:

<img width="999" alt="Screenshot 2024-03-21 at 10 13 33 AM" src="https://github.com/ionic-team/stencil-site/assets/6207644/29b65233-ef06-4a03-8287-b8ca6dd3af35">

Maybe I did something wrong! Probably!

I only have one test file I created which looks like this:

```ts
import { expect } from '@playwright/test';
import { test } from '@stencil/playwright';

test.describe('my-component', () => {
  test('should render the correct name', async ({ page }) => {
    // Assume we have a template setup with the `my-component` component and a `button`
    // await page.goto('/my-component/my-component.e2e.html');
    await page.setContent('<my-component first="Stencil"></my-component>');

    const button = page.locator('button');
    // Assmume clicking the button changes the `@Prop()` values on `my-component`
    button.click();
    await page.waitForChanges();
  });
});
```

it's saved in `src/my-component/my-component.e2e.ts`

Reply:
@alicewriteswrongs So this is one bit I don't like, this is caused by the `Symbol.asyncDispose` issue. I bet if you run `npm run build`, you'll see an error. The problem is that Playwright (or our implementation), doesn't report build errors when running the dev server, so nothing fails until the timeout is reached.

I have a task in Jira to figure out a way for us to log those errors and "fail quick" when they happen.

For now, I'll update the step below this one to always recommend adding the `ESNext.Disposable` property in the `tsconfig.json` since this is a problem with every starter project. In fact, I'll probably move it up in this list.